### PR TITLE
fix #8377 feat(cirrus): Initial commit

### LINKED ISSUE #8377: Set up Cirrus and Fast API
Since we are adding Cirrus service to support non-firefox applications, set up cirrus in the root folder of the experimenter repo as a subfolder server and client. In a server set up FastApi with the docker file.
┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-3211)

### PR DESCRIPTION
**Pull request by yashikakhurana**
Because

- We want to add Cirrus to the experimenter repo

This commit

-  Set up the cirrus server and requirements
-  Set up Dockerfile and include cirrus in docker-compose file
-  To run the cirrus server- use command `make cirrus_up`

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -193,3 +193,11 @@ integration_test_nimbus: build_prod
...
 integration_test_nimbus_rust: build_prod
 	MOZ_HEADLESS=1 $(COMPOSE_INTEGRATION) run rust-sdk sh -c "chmod a+rwx /code/app/tests/integration/.tox;tox -c app/tests/integration -e integration-test-nimbus-rust $(TOX_ARGS) -- -n 2 $(PYTEST_ARGS)"
+
+
+
+cirrus_up:  ## Run Docker Compose services
+	$(COMPOSE) up web
+
+cirrus_down:  ## Shutdown Docker Compose services
+	$(COMPOSE) down
```
Comment (jeddai) [inline Makefile:203]: Can we reference the docker compose service so it only shuts down Cirrus?

File: docker-compose.yml [hunk 2]
```diff
@@ -101,6 +101,16 @@ services:
     ports:
       - "8000:8000"
 
+  web:
```
Comment (jeddai) [inline docker-compose.yml:104]: Considering this is in a compose with a number of other web services, would it maybe make sense for the service to be called Cirrus?

**Review by jeddai — COMMENTED** [review]
Just a couple thoughts but this is looking good!

File: docker-compose.yml (hunk 2 above)
Comment (yashikakhurana) [reply docker-compose.yml]: good call

File: docker-compose.yml [hunk 3]
```diff
@@ -101,6 +101,16 @@ services:
...
       - "8000:8000"
 
+  cirrus:
+    build:
+      context: .
+      dockerfile: cirrus/server/Dockerfile
+    volumes:
+      - .:/cirrus/server
+    ports:
+      - "8000:8000"
```
Comment (jaredlockhart) [inline docker-compose.yml:111]: 8000 is already in use by autograph above?

**Review by jaredlockhart — DISMISSED** [review]
Yay Cirrus! 🎉

**Review by jeddai — APPROVED** [review]
looking great 🎉

**Pull request merged**